streams/src/test/java/org/apache/kafka/streams/integration/IQv2VersionedStoreIntegrationTest.java

```diff
@@ -0,0 +1,198 @@
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Category({IntegrationTest.class})
+public class IQv2VersionedStoreIntegrationTest {
+    private static final int NUM_BROKERS = 1;
+    private static final String INPUT_TOPIC_NAME = "input-topic";
+    private static final String STORE_NAME = "versioned-store";
+    private static final Duration HISTORY_RETENTION = Duration.ofDays(1);
+    private static final Instant RECORD_TIMESTAMP = Instant.now();
+    private static final Long RECORD_TIMESTAMP_LONG = RECORD_TIMESTAMP.toEpochMilli();
+
+
+    private static final Logger LOG = LoggerFactory.getLogger(IQv2VersionedStoreIntegrationTest.class);
+
+    public static final EmbeddedKafkaCluster CLUSTER = new EmbeddedKafkaCluster(NUM_BROKERS,
+        Utils.mkProperties(Collections.singletonMap("auto.create.topics.enable", "true")));
+    private StreamsBuilder builder;
+    private KafkaStreams kafkaStreams;
+
+    public IQv2VersionedStoreIntegrationTest(){}
+
+    @BeforeClass
+    public static void before()
+        throws InterruptedException, IOException, ExecutionException, TimeoutException {
+        CLUSTER.start();
+        final Properties producerProps = new Properties();
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
+        Thread.sleep(10000);
+        final KafkaProducer<Integer, Integer> producer = new KafkaProducer<>(producerProps);
+        producer.send(new ProducerRecord<>(INPUT_TOPIC_NAME, 0, RECORD_TIMESTAMP_LONG, 2, 2)).get();
+    }
+
+    @Before
+    public void beforeTest() throws InterruptedException {
+        builder = new StreamsBuilder();
+        builder.table(INPUT_TOPIC_NAME,
+            Materialized.as(Stores.persistentVersionedKeyValueStore(STORE_NAME, HISTORY_RETENTION)));
+        final Properties configs = new Properties();
+        configs.put(StreamsConfig.APPLICATION_ID_CONFIG, "app");
+        configs.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        configs.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.IntegerSerde.class.getName());
+        configs.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.IntegerSerde.class.getName());
+        kafkaStreams = new KafkaStreams(builder.build(), configs);
+        kafkaStreams.start();
+        Thread.sleep(10000);
+    }
+
+    @After
+    public void afterTest() {
+        if (kafkaStreams != null) {
+            kafkaStreams.close();
+            kafkaStreams.cleanUp();
+        }
+    }
+
+    @AfterClass
+    public static void after() {
+        CLUSTER.stop();
+    }
+
+    @Test
+    public void verifyStore() {
```

Comment:
Seems this is testing three different things -- should we make three test methods instead?

Reply:
> Seems this is testing three different things -- should we make three test methods instead?

Then the `beforeTest` method is executed two times.

Reply:
Yes, but that seems ok to me? -- It's always a judgment call -- I am also ok to leave it as-is (in doubt, I prefer to split).